rustls/src/webpki/verify.rs

```diff
@@ -125,6 +127,13 @@ impl fmt::Debug for WebPkiSupportedAlgorithms {
 /// This is used in order to avoid parsing twice when specifying custom verification
 pub struct ParsedCertificate<'a>(pub(crate) webpki::EndEntityCert<'a>);
 
+impl<'a> ParsedCertificate<'a> {
+    /// Get the parsed certificate's SubjectPublicKeyInfo (SPKI)
+    pub fn subject_public_key_info(&self) -> SubjectPublicKeyInfoDer {
```

Comment:
Nit: would be good to make the lifetime in the return value explicit here, `<'static>`.

Reply:
Can you explain why we'd want to do this? Plus how to do this. I'm less solid on lifetimes.

* **Why** I'm not sure I understand:
  * The motivation for making the lifetime API explicit instead of elided. Is this because we expect (by inspection) the return value to be always `'static`, and so we might as well make it explicit?
  * Why the return value for this function would always be `'static`, and why we'd want that
* **How** This is how I'd naively do this. I'm unsure how to appease the borrow checker without tacking on an extra `into_owned` (which looks like it might incur an extra copy):

  ```diff
   impl<'a> ParsedCertificate<'a> {
       /// Get the parsed certificate's SubjectPublicKeyInfo (SPKI)
  -    pub fn subject_public_key_info(&self) -> SubjectPublicKeyInfoDer {
  -        self.0.subject_public_key_info()
  +    pub fn subject_public_key_info(self) -> SubjectPublicKeyInfoDer<'static> {
  +       self.0
  +            .subject_public_key_info()
  +            .into_owned()
      }
  }
  ```

Reply:
There's no need to do `into_owned()` because the webpki `subject_public_key_info()` method always yields a `SubjectPublicKeyInfoDer<'static>`.

Reply:
> The motivation for making the lifetime API explicit instead of elided.

I've seen experienced rustaceans point to this practice as a debugging aid (e.g. https://quinedot.github.io/rust-learning/dont-hide.html).

Reply:
> (See https://github.com/rustls/webpki/commit/211550edfdb8fe2b0d03760acc30fede55874f28.)

Ahhh ok I see what we're missing. Your suggested change doesn't compile without your linked webpki commit that updates the `subject_public_key_info` signature to explicitly add that `'static` lifetime. I was able to confirm this by pointing Rustls' Cargo to webpki's main branch and compiling everything without a hitch. I believe this means we have to:

1. Put that commit into a webpki release
2. Update Rustls to use that release

Compiler error without the linked commit for convenience:
```rustc
error: lifetime may not live long enough
   --> rustls/src/webpki/verify.rs:133:9
    |
130 | impl<'a> ParsedCertificate<'a> {
    |      -- lifetime `'a` defined here
...
133 |         self.0.subject_public_key_info()
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ returning this value requires that `'a` must outlive `'static`

error: lifetime may not live long enough
   --> rustls/src/webpki/verify.rs:133:9
    |
132 |     pub fn subject_public_key_info(&self) -> SubjectPublicKeyInfoDer<'static> {
    |                                    - let's call the lifetime of this reference `'1`
133 |         self.0.subject_public_key_info()
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ returning this value requires that `'1` must outlive `'static`

help: replace `'a` with `'static`

error: could not compile `rustls` (lib) due to 2 previous errors
```

Reply:
^ Bump. Am I missing something here?

Reply:
You're correct :-) We'll get the webpki release published shortly and then this should be good to go.

Reply:
This has been released: https://github.com/rustls/webpki/releases/tag/v%2F0.102.5. Make sure to update the requirement to 0.102.5 in `Cargo.toml`.

Reply:
Updated!